### ISSUE TITLE
"No Response" when defining multiple Avea bulbs

### ISSUE THREAD

**Issue by hezi** [kounch/homebridge-avea-bulb#6]
Hi!
I'm trying to configure my homebridge with two avea bulbs, unfortunately, when i do so the home app on my phone doesn't get any response from the bulbs. homebridge log shows nothing, just the two bulb being found and scan stopping.

my accessory config looks like this:
```json
  "accessories": [
    {
      "name":"left light",
      "accessory": "AveaBulb",
      "bluetoothid": "84eb186ab923"
    },
    {
      "name":"right light",
      "accessory": "AveaBulb",
      "bluetoothid": "f4b85e9597ec"
    }
]
``` 

and the homebridge log looks like this:
```
[2018-2-3 07:01:41] Homebridge is running on port 51826.
[2018-2-3 07:01:41] Pi Camera is running on port 39494.
[2018-2-3 07:01:41] [left light] Starting Noble scan..
[2018-2-3 07:01:41] [right light] Starting Noble scan..
[2018-2-3 07:01:41] [left light] Discovered: f4b85e9597ec
[2018-2-3 07:01:41] [right light] Discovered: f4b85e9597ec
[2018-2-3 07:01:41] [left light] Discovered: 84eb186ab923
[2018-2-3 07:01:41] [right light] Discovered: 84eb186ab923
```

With no other avea related messages down the line.
When I only configure one bulb things seem to work properly. 

Thanks!

**Comment by kounch** [comment]
I am trying to reproduce the issue but, since i only have now one bulb, i cannot set it exactly the same way as you have.
Setting up two bulbs (a real one and a fake one) is working properly for me.

Could you try to add this lines to your `index.js` file in `.../node_modules/homebridge-avea-bulb/index.js`?

Changing from line 112 on as follows.

Old code:
```
this.bulb.connect();
            }
        } else {
            // do a reconnect if uuid matches
            if (peripheral.uuid == this.bluetoothid) {
                this.log("Lost bulb appears again!");
                this.perifSel = peripheral;
                if (this.perifSel.state != "connected") {
                    Noble.stopScanning();
                    this.scanning = false;
                    this.bulb = new AveaBulb.Avea(this.perifSel);
                    this.bulb.connect();
                } else {
                    this.log("Undefined state");
                }
            }
```

New test code:
```
this.bulb.connect();
            } else {
                this.log("UUID not matching");
            }
        } else {
            // do a reconnect if uuid matches
            if (peripheral.uuid == this.bluetoothid) {
                this.log("Lost bulb appears again!");
                this.perifSel = peripheral;
                if (this.perifSel.state != "connected") {
                    Noble.stopScanning();
                    this.scanning = false;
                    this.bulb = new AveaBulb.Avea(this.perifSel);
                    this.bulb.connect();
                } else {
                    this.log("Undefined state");    
                }
            } else {
                this.log("This is not the bulb you are looking for");
            }
        }
```

...and then report here what you see in the log file?

Thanks

**Comment by hezi** [comment]
This is the result I get with the debug prints added: 
```
[2018-2-3 16:59:10] Homebridge is running on port 51826.
[2018-2-3 16:59:10] Pi Camera is running on port 42273.
[2018-2-3 16:59:11] [left light] Starting Noble scan..
[2018-2-3 16:59:11] [right light] Starting Noble scan..
[2018-2-3 17:00:05] [left light] Discovered: f4b85e9597ec
[2018-2-3 17:00:05] [left light] UUID not matching
[2018-2-3 17:00:05] [right light] Discovered: f4b85e9597ec
[2018-2-3 17:00:05] [right light] UUID matches!
[2018-2-3 17:00:05] [left light] ScanStop received
[2018-2-3 17:00:05] [right light] ScanStop received
[2018-2-3 17:00:05] [left light] ScanStop received
[2018-2-3 17:00:05] [right light] ScanStop received
[2018-2-3 17:00:05] [left light] Discovered: 84eb186ab923
[2018-2-3 17:00:05] [left light] UUID matches!
[2018-2-3 17:00:05] [right light] Discovered: 84eb186ab923
[2018-2-3 17:00:05] [right light] This is not the bulb you are looking for
[2018-2-3 17:00:05] [left light] ScanStop received
[2018-2-3 17:00:05] [right light] ScanStop received
[2018-2-3 17:00:05] [left light] ScanStop received
[2018-2-3 17:00:05] [right light] ScanStop received
noble warning: unknown handle 65 disconnected!
noble warning: unknown handle 65 disconnected!
noble warning: unknown handle 65 disconnected!
```

**Comment by kounch** [comment]
What hardware is your homebridge installation. Are you using a Raspberry Pi 3 and you are connected to Wifi too?

**Comment by hezi** [comment]
I'm using a Raspberry Pi Zero W

**Comment by kounch** [comment]
Then it looks like a known hardware problem :( as seen here: sandeepmistry/noble#465, sandeepmistry/noble#474, sandeepmistry/noble#486. 

Using Bluetooth when connected to Wifi will interfere with Wifi, causing disconnects and eventually hangups. To solve this, one option they recommend is to disable Wifi and go with a wired solution, or getting an external Bluetooth adapter.

In your case, it appears that the extra bluetooth traffic of using a second bulb triggers the error.

I have found here: zwaldowski/homebridge-satellite-fan#1 that they tried different strategies to try to wok around it. I will look into their code and see if i can add them to mine.

**Comment by hezi** [comment]
Thanks for the information, highly appreciated. I'll take a look at their workaround too and see if i can offer a pull request.

**Comment by kounch** [comment]
There is a new testing version with some new connection strategies and more debugging messages. 

You can replace your current version with this new one for testing with:
```
npm uninstall -g homebridge-avea-bulb
npm install -g --unsafe-perm git+https://github.com/kounch/homebridge-avea-bulb.git#develop
```

And come back here with your results.

**Comment by hezi** [comment]
I've been testing it for a while now, and apart from the occasional "Lamp not responding" in the Home app (which goes away after a minute) it seems to be working. Thanks! :)

**Comment by kounch** [comment]
Included in release 1.1.0

Closed
